Sources/eppo/RuleEvaluator.swift

```diff
@@ -175,95 +240,131 @@ public class FlagEvaluator {
+                default:
+                    return false
+                }
+            } else {
+                // If either string is not a valid Semver, fall back to double comparison
+                guard let valueDouble = try? value.getDoubleValue() else {
+                    return false
+                }
+                
+                // If the config is obfuscated, we need to unobfuscate the condition value
+                var conditionDouble: Double
+                if isConfigObfuscated,
+                   let cvs = conditionValueStr,
+                   let doubleValue = Double(cvs) {
+                    conditionDouble = doubleValue
+                } else if let doubleValue = try? condition.value.getDoubleValue() {
+                    conditionDouble = doubleValue
+                } else {
+                    return false
+                }
+                
+                switch condition.operator {
+                case .greaterThanEqual:
+                    return valueDouble >= conditionDouble
+                case .greaterThan:
+                    return valueDouble > conditionDouble
+                case .lessThanEqual:
+                    return valueDouble <= conditionDouble
+                case .lessThan:
+                    return valueDouble < conditionDouble
+                default:
                     return false
                 }
-            case .matches:
-                return Compare.matchesRegex(
-                    try value.toEppoString(),
-                    try condition.value.toEppoString()
-                )
-            case .notMatches:
-                return !Compare.matchesRegex(
-                    try value.toEppoString(),
-                    try condition.value.toEppoString()
-                )
-            case .oneOf:
-                return try Compare.isOneOf(
-                    value.toEppoString(),
-                    condition.value.getStringArrayValue()
-                )
-            case .notOneOf:
-                return try !Compare.isOneOf(
-                    value.toEppoString(),
-                    condition.value.getStringArrayValue()
-                )
-            default:
-                return false;
             }
-        } catch {
-            // Handle or log the error
+        case .matches, .notMatches:
+            if let conditionString = try? condition.value.toEppoString(),
+               let valueString = try? value.toEppoString() {
+                
+                if isConfigObfuscated,
+                   let decoded = base64Decode(conditionString) {
+                    return condition.operator == .matches ? Compare.matchesRegex(valueString, decoded) : !Compare.matchesRegex(valueString, decoded)
+                } else {
+                    return condition.operator == .matches ? Compare.matchesRegex(valueString, conditionString) : !Compare.matchesRegex(valueString, conditionString)
```

Comment:
could we just init decoded to value string, and then decode if obfuscated, and then only have one of these returns?

Here is an example in the [Android SDK](https://github.com/Eppo-exp/android-sdk/blob/main/eppo/src/main/java/cloud/eppo/android/RuleEvaluator.java#L175)